src/VisualStudio/Core/Def/Implementation/SymbolSearch/SymbolSearchResult.cs

```diff
@@ -0,0 +1,232 @@
+            result.DocumentGuid = documentGuid.ToString();
+            result.ProjectName = projectName;
+
+            if (referenceItem != null)
+            {
+                var classifiedSpansAndHighlightSpan = await ClassifiedSpansAndHighlightSpanFactory.ClassifyAsync(referenceItem.SourceSpan, cancellationToken).ConfigureAwait(false);
+                var classifiedSpans = classifiedSpansAndHighlightSpan.ClassifiedSpans;
+                var docText = await referenceItem.SourceSpan.Document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+                result.ContextText = new ClassifiedTextElement(classifiedSpans.Select(cspan => new ClassifiedTextRun(cspan.ClassificationType, docText.ToString(cspan.TextSpan))));
+                result.ContextHighlightSpan = AsSpan(classifiedSpansAndHighlightSpan.HighlightSpan);
+
+                if (referenceItem.AdditionalProperties.TryGetValue(AbstractReferenceFinder.ContainingTypeInfoPropertyName, out var containingTypeInfo))
+                {
+                    result.ContainingTypeName = containingTypeInfo;
+                }
+
+                if (referenceItem.AdditionalProperties.TryGetValue(AbstractReferenceFinder.ContainingMemberInfoPropertyName, out var containingMemberInfo))
+                {
+                    result.ContainingMemberName = containingMemberInfo;
+                }
+
+                // Get read\written information
+                if (referenceItem.IsWrittenTo)
+                {
+                    result.IsWrittenTo = true;
+                }
+                else
+                {
+                    result.IsReadFrom = true;
+                }
+
+                // Override read\writtern information from SymbolUsageInfo
+                if (referenceItem.SymbolUsageInfo.IsWrittenTo() && referenceItem.SymbolUsageInfo.IsReadFrom())
+                {
+                    result.IsReadFrom = true;
+                    result.IsWrittenTo = true;
+                }
+                else if (referenceItem.SymbolUsageInfo.IsWrittenTo())
+                {
+                    result.IsReadFrom = false;
+                    result.IsWrittenTo = true;
+                }
+                else if (referenceItem.SymbolUsageInfo.IsReadFrom())
+                {
+                    result.IsReadFrom = true;
+                    result.IsWrittenTo = false;
+                }
+
+                var definitionResult = context.GetDefinitionResult(referenceItem.Definition);
+                if (definitionResult != null)
+                {
+                    result.Definition = definitionResult;
+                }
+                else
+                {
+                    System.Diagnostics.Debugger.Break();
+                    // I assumed that definitions are profferred first, but we just received a reference before getting a definition.
+                    // In this case, create a queue of definitions that need to have their Results realized.
```

Comment:
how can this happen?  there is something *very* bad if you are hititng this.

Reply:
See comment in OnDefinitionFoundAsync -- it looks like it doesn't add it in case of the definition having multiple source spans. I wonder if that's what's happening here.